### PR TITLE
Refactor ApplicationExitInfo to move it to Event.Application.Execution instead of Report.

### PR DESCRIPTION
**Pull request by tejasd**

### CODE REVIEW

File: firebase-crashlytics/src/test/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReportTest.java [hunk 1]
```diff
@@ -226,16 +231,42 @@ private static Application makeTestApplication() {
   }
 
   private static Event makeAnrEvent() {
-    return makeTestEvent("ANR");
+    return Event.builder()
```
Comment (pydichandra) [inline firebase-crashlytics/src/test/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReportTest.java:234]: I see similar create ANR event methods in other unit tests as well. would it be possible to create a utility method to create ANR and use that in all the tests?

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCaptureTest.java [hunk 2]
```diff
@@ -101,6 +102,21 @@ public void testCaptureReport_containsNoDeveloperPlatformFieldsWhenUnityIsMissin
...
+    CrashlyticsReport.ApplicationExitInfo testApplicationExitInfo = makeAppExitInfo();
+    final CrashlyticsReport.Session.Event event =
+        dataCapture.captureAnrEventData(testApplicationExitInfo);
+
+    assertEquals("anr", event.getType());
+    assertEquals(testApplicationExitInfo, event.getApp().getExecution().getAppExitInfo());
+    assertEquals(testApplicationExitInfo.getTimestamp(), event.getTimestamp());
+    assertEquals(
+        testApplicationExitInfo.getImportance()
+            != ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND,
```
[the remark this answers is not in the document]
Comment (tejasd) [reply firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCaptureTest.java:116]: Cleaned it up to make it more obvious (and added an extra test), but yes, converting an AppExitInfo to Event uses testApplicationExitInfo.getImportance() to decide whether it's foreground or background.

File: firebase-crashlytics/src/test/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReportTest.java (hunk 1 above)
Comment (tejasd) [reply firebase-crashlytics/src/test/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReportTest.java:234]: The codebase currently doesn't use a utility class and there are a decent number of methods that are repeated across unit tests eg.

1. makeTestApplication
2. makeTestEvents

etc.

I can take on a task to refactor this in a separate PR!

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java [hunk 3]
```diff
@@ -110,6 +110,17 @@ public Event captureEventData(
         .build();
   }
 
+  public Event captureAnrEventData(CrashlyticsReport.ApplicationExitInfo applicationExitInfo) {
+    final int orientation = context.getResources().getConfiguration().orientation;
```
[the remark this answers is not in the document]
Comment (tejasd) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsReportDataCapture.java:117]: Done.

**Pull request merged**